### ISSUE TITLE
go test ./... fails to run

### ISSUE THREAD

**Issue by pxp928** [in-toto/in-toto-golang#154]
**General information:**
make test fails with the new test added in provenance_test.go. `BuildStartedOn` and `BuildFinishedOn` are hardcoded into the `TestEncodeProvenancePredicate` causing it to fail.  

**Description of the bug:**
```
?       github.com/in-toto/in-toto-golang       [no test files]
?       github.com/in-toto/in-toto-golang/cmd   [no test files]
ok      github.com/in-toto/in-toto-golang/in_toto       (cached)
--- FAIL: TestEncodeProvenancePredicate (0.00s)
    provenance_test.go:122:   strings.Join({
                ... // 122 identical bytes
                `ns/GitHubActionsWorkflow@v1","definedInMaterial":0,"entryPoint":`,
                `"build.yaml:maketgz"},"metadata":{"buildStartedOn":"2020-08-19T0`,
        -       "8:38:00Z",
        +       "4:38:00-04:00",
                `","buildFinishedOn":"2020-08-19T`,
        -       "08:38:00Z",
        +       "04:38:00-04:00",
                `","completeness":{"arguments":true,"environment":false,"material`,
                `s":true},"reproducible":false},"materials":[{"uri":"git+https://`,
                ... // 189 identical bytes
          }, "")
        
FAIL
FAIL    github.com/in-toto/in-toto-golang/in_toto/slsa_provenance/v0.1  0.154s
ok      github.com/in-toto/in-toto-golang/internal/spiffe       (cached)
?       github.com/in-toto/in-toto-golang/internal/test [no test files]
FAIL
make: *** [go-test] Error 1
```

**Comment by adityasaky** [comment]
This seems like a timezone issue. The test uses a predetermined time defined here: https://github.com/in-toto/in-toto-golang/blob/master/in_toto/slsa_provenance/v0.1/provenance_test.go#L39. I agree, it should be handled.

**Comment by pxp928** [comment]
Line 39 does not seem to be an issue. The issue is with:

https://github.com/in-toto/in-toto-golang/blob/master/in_toto/slsa_provenance/v0.1/provenance_test.go#L118

**Comment by adityasaky** [comment]
Ah right, I missed the location was handled further down. Unmarshallin want and using it to set the location should fix it I think?

**Comment by pxp928** [comment]
The test is comparing the JSON output so would not invalidate that test?
```
assert.Equal(t, want, string(b), "Wrong JSON produced")
```

**Comment by adityasaky** [comment]
```
	testTime = testTime.In(time.UTC)
```

This is what I had in mind, right after defining `testTime` using `time.Unix`?

**Comment by adityasaky** [comment]
Closed by #155.
